### PR TITLE
Remove JMX receiver

### PR DESCRIPTION
**Pull request by flands**
Leverage SA library for now. Also run go mod tidy.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -18,13 +21,11 @@ require (
...
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/collectdreceiver v0.19.1-0.20210203200406-65673ad3657b
-	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/dockerstatsreceiver v0.19.1-0.20210203200406-65673ad3657b
-	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/jmxreceiver v0.19.1-0.20210203200406-65673ad3657b
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver v0.19.1-0.20210203200406-65673ad3657b
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/kubeletstatsreceiver v0.19.1-0.20210203200406-65673ad3657b
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusexecreceiver v0.19.1-0.20210203200406-65673ad3657b
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/receivercreator v0.19.1-0.20210203200406-65673ad3657b
-	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/redisreceiver v0.19.1-0.20210203200406-65673ad3657b
+	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/redisreceiver v0.19.1-0.20210203200406-65673ad3657b // indirect
```
Comment (jrcamp) [inline go.mod:28]: Wait how did this get added back?

Comment (jrcamp) [reply go.mod:28]: Oh was removed from https://github.com/signalfx/splunk-otel-collector/pull/84/files but not go.mod. Hopefully can be removed?

**Review by rmfitzpatrick — APPROVED** [review]
lgtm w/ redis receiver removed and conflicts updated

File: go.mod (hunk 1 above)
Comment (flands) [reply go.mod:28]: kubletstats requires redis

```
go: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/kubeletstatsreceiver@v0.19.1-0.20210203200406-65673ad3657b requires
	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/redisreceiver@v0.0.0-00010101000000-000000000000
```

**Pull request merged**